kedro-datasets/tests/pandas/test_sql_dataset.py

```diff
@@ -419,3 +420,37 @@ def test_create_connection_only_once(self, mocker):
         assert mock_engine.call_count == 2
         assert fourth.engines == first.engines
         assert len(first.engines) == 2
+
+    def test_adapt_mssql_date_params_called(self, mocker):
+        """Test that the adapt_mssql_date_params
+        function is called when mssql backend is used.
+        """
+        mock_adapt_mssql_date_params = mocker.patch(
+            "kedro_datasets.pandas.sql_dataset.SQLQueryDataSet.adapt_mssql_date_params"
+        )
+        mock_engine = mocker.patch("kedro_datasets.pandas.sql_dataset.create_engine")
+        ds = SQLQueryDataSet(sql=SQL_QUERY, credentials=dict(con=MSSQL_CONNECTION))
+        mock_engine.assert_called_once_with(MSSQL_CONNECTION)
+        assert mock_adapt_mssql_date_params.call_count == 1
+        assert len(ds.engines) == 1
+
+    def test_adapt_mssql_date_params(self):
+        """Test that the adapt_mssql_date_params
+        function transforms the params as expected, i.e.
+        making datetime date into the format %Y-%m-%dT%H:%M:%S
+        and ignoring the other values.
+        """
+        load_args = {
+            "params": ["2023-01-01", "2023-01-01T20:26", "2023", "test", 1.0, 100]
+        }
+        ds = SQLQueryDataSet(
```

Comment:
The unit tests are failing as the init method makes a call to `create_connection()` (and in turn `create_engine()`) which results in an import error. I suspect you are missing a mock_engine as you've used in the test above 🤔

Reply:
Oh right, that's a good catch, thanks. It was working locally so I forgot about the mock, thanks for pointing this. :+1: